packages/generator/src/renderer.ts

```diff
@@ -282,6 +299,38 @@ function renderApiFamily(
   return apiNames;
 }
 
+/**
+ * Read all the RAML files for an API family and process into AML models.
+ *
+ * @param apiFamily - The name of the API family
+ * @param apiFamilyConfig - The API family config
+ * @param inputDir - The path to read the RAML files from
+ *
+ * @returns a list of promises that will resolve to the AMF models
+ */
+export function processApiFamily(
+  apiFamily: string,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  apiFamilyConfig: any,
+  inputDir: string
+): Promise<model.document.BaseUnit>[] {
+  const promises = [];
+  const ramlFileFromFamily = apiFamilyConfig[apiFamily];
+  _.map(ramlFileFromFamily, (apiMeta: RestApi) => {
+    if (!apiMeta.id) {
+      throw Error(`Some information about '${apiMeta.name}' is missing in 'apis/api-config.json'. 
+      Please ensure that '${apiMeta.name}' RAML and its dependencies are present in 'apis/', and all the required information is present in 'apis/api-config.json'.`);
+    }
+    promises.push(
+      parseRamlFile(
+        path.join(inputDir, apiMeta.assetId, apiMeta.fatRaml.mainFile)
+      )
+    );
+  });
+
+  return promises;
```

Comment:
does it make sense to wait for all the promises to resolve and then return the array of apis instead of promises

Reply:
My current work also touches this function, and I have already fixed it on my branch. You can leave this for now.